### ISSUE TITLE
HTTP Endpoints/URLs should be consistent regardine path vs. query arguments

### ISSUE THREAD

**Issue by sboagy** [sboagy/tunetrees#110]
I started with paths, for example `/tunetrees/scheduled_tunes_overview/{user_id}/{playlist_ref}`, and then at one point decided I should be using query parameters, so a bunch of the endpoints are using query parameters now.  But now I realize they should have been paths after all, so I was right the first time?  I still get a little fuzzy on the hierarchical nature of `/tunetrees/scheduled_tunes_overview/` vs. `user_id` and `playlist_ref` seems like a "filter" similar to `genre=rock`.  But, what's important is convention and consistency.

I may see if I can get CoPilot to do the refactor for me.

==== gemini ====

> If I have a endpoint like `/tunetrees/scheduled_tunes_overview/{user_id}/{playlist_ref}` for a row in a table, explain to me why the URL should use paths vs. query parameters for {user_id} etc.

```
/tunetrees/scheduled_tunes_overview/{user_id}/{playlist_ref}
```

using path parameters (like `{user_id}` and `{playlist_ref}`) instead of query parameters offers several advantages in terms of RESTful API design and clarity.

**Advantages of Path Parameters**

1. **Resource Identification**

   * **Clear hierarchy:** Path parameters are used to identify specific resources within a hierarchical structure. In your example, `{user_id}` and `{playlist_ref}` are essential for identifying the specific scheduled tunes overview you want to access.
   * **RESTful convention:**  This aligns with RESTful principles, where URLs represent resources, and path segments indicate the hierarchy of those resources.

2. **Clarity and Readability**

   * **Meaningful URLs:** Path parameters make the URL more meaningful and easier to understand. For example, `/tunetrees/scheduled_tunes_overview/123/456` clearly indicates that you're accessing the scheduled tunes overview for user 123 and playlist 456.
   * **Less clutter:** Using path parameters avoids cluttering the URL with query parameters, making it more concise and readable.

3. **Caching**

   * **Cacheable resources:** URLs with path parameters are often more cacheable than those with query parameters. This can improve performance by allowing clients or intermediate servers to cache the responses for specific resources.

4. **API Structure**

   * **Resource-oriented design:** Path parameters emphasize the resource-oriented nature of your API. Each path segment represents a specific resource or collection of resources.
   * **Consistency:** Using path parameters consistently for resource identification makes your API more predictable and easier to use.

**When to Use Query Parameters**

Query parameters are more suitable for:

* **Filtering:**  `/tunetrees/tunes?genre=rock`
* **Sorting:** `/tunetrees/tunes?sort=title`
* **Pagination:** `/tunetrees/tunes?page=2&limit=10`
* **Optional parameters:** When a parameter is not essential for identifying the resource.

**In summary**

For your `/tunetrees/scheduled_tunes_overview/{user_id}/{playlist_ref}` endpoint, using path parameters is the preferred approach because it clearly identifies the specific resource within a hierarchy, aligns with RESTful conventions, and improves URL readability and cacheability.
